src/auditlog/backends/base.py

```diff
@@ -0,0 +1,51 @@
+from auditlog.diff import model_instance_diff
+from auditlog.models import LogEntry
+
+
+class AuditBackend(object):
+    """Base backend for creation of audit log entries."""
+
+    def log_create(self, instance, created, **kwargs):
+        """Handle model creation logging"""
+        if created:
+            changes = model_instance_diff(None, instance)
+
+            self.create_log(
+                instance=instance,
+                action=LogEntry.Action.CREATE,
+                changes=changes,
+                **kwargs)
+
+    def log_update(self, instance, **kwargs):
+        """Handle model update logging"""
+        if instance.pk is not None:
+            try:
+                old = instance.__class__.objects.get(pk=instance.pk)
+            except instance.__class__.DoesNotExist:
+                pass
+            else:
+                new = instance
+
+                changes = model_instance_diff(old, new)
+
+                # Log an entry only if there are changes
+                if changes:
+                    self.create_log(
+                        instance=instance,
+                        action=LogEntry.Action.UPDATE,
+                        changes=changes,
+                        **kwargs)
+
+    def log_delete(self, instance, **kwargs):
+        """Handle model deletion logging"""
+        if instance.pk is not None:
+            changes = model_instance_diff(instance, None)
+
+            self.create_log(
+                instance=instance,
+                action=LogEntry.Action.DELETE,
+                changes=changes,
+                **kwargs)
+
+    def create_log(self, action, instance, changes, **kwargs):
```

Comment:
Confusing names: log_create and create_log are too similar while both belong to public interface. Adding a leading underscore to create_log or a name change should help.

Reply:
I would tend to agree, but this way you can skip the signals and call `create_log` directly without having to go through model signals.

Reply:
So we can just change the name to be more meaningful and distinguishable.